core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -30,145 +23,35 @@
  *
  * <p>For example, {@code TagContext}s can be used to label stats, log messages, or debugging
  * information.
+ *
+ * <p>Keys have type {@link TagKey}. Values have type {@link TagValueString}, though the library
+ * will support more types in the future, including {@code long} and {@code boolean}.
  */
+// TODO(sebright): Consider removing TagContext.iterator() so that we don't need to support fast
+// access to tags.
 @Immutable
-public final class TagContext {
-
-  /** The empty {@code TagContext}. */
-  public static final TagContext EMPTY = emptyBuilder().build();
-
-  // The types of the TagKey and value must match for each entry.
-  private final Map<TagKey, Object> tags;
-
-  TagContext(Map<? extends TagKey, ?> tags) {
-    this.tags = Collections.unmodifiableMap(new HashMap<TagKey, Object>(tags));
-  }
-
-  Map<TagKey, Object> getTags() {
-    return tags;
-  }
+public abstract class TagContext implements Iterable<Tag> {
```

Comment:
I changed it to `unsafeGetIterator()` for now and left the TODOs about possibly removing the ability to iterate over the tags.